src/autowiring/AutoPacket.cpp

```diff
@@ -274,6 +271,50 @@ void AutoPacket::UpdateSatisfactionUnsafe(std::unique_lock<std::mutex> lk, const
 
   // Recursively mark unsatisfiable any single-output arguments on these subscribers:
   std::vector<const AutoFilterArgument*> unsatOutputArgs;
+  auto MarkOutputsUnsat = [&unsatOutputArgs] (const SatCounter& satCounter) {
+    const auto* args = satCounter.GetAutoFilterArguments();
+    for (size_t i = satCounter.GetArity(); i--;) {
+      // Only consider output arguments:
+      if (args[i].is_output)
+        // This output is transitively unsatisfiable, include it for later removal
+        unsatOutputArgs.push_back(&args[i]);
+    }
+  };
+
+  if (!disposition.m_modifiers.empty() && disposition.m_decorations.size() > 1)
+    throw autowiring_error("An AutoFilter was detected which has single-decorate rvalue argument in a graph with multi-decorate outputs");
```

Comment:
Added the checking to `AddSatCounterUnsafe`. It will be called when the packet is created and also when the packet has a lambda attached to it.
For our current implementation, don't think it is possible to detect this condition on the AutoPacketFactory itself, because when adding an autofilter, all it does is insert it to a linked list. We only construct the graph when a new packet is created.